src/ImageRunModal.jsx

```diff
@@ -351,68 +353,116 @@ export class ImageRunModal extends React.Component {
-                this.activeConnection.call({
+        const searches = [];
+
+        // Try to get specified image manifest
+        searches.push(this.activeConnection.call({
+            method: "GET",
+            path: client.VERSION + "libpod/manifests/" + value + "/json",
+            body: "",
+        }));
+
+        // Don't start search queries when tag is specified as search API doesn't support
+        // searching for specific image tags
+        // instead only rely on manifests query (requires image:tag name)
+        if (!RE_CONTAINER_TAG.test(value)) {
+            // If there are registries configured search in them, or if a user searches for `docker.io/cockpit` let
+            // podman search in the user specified registry.
+            if (Object.keys(this.props.podmanInfo.registries).length !== 0 || value.includes('/')) {
+                searches.push(this.activeConnection.call({
                     method: "GET",
                     path: client.VERSION + "libpod/images/search",
                     body: "",
                     params: {
-                        term: registry + "/" + value
+                        term: value,
                     }
-                })));
+                }));
+            } else {
+                searches.push(...utils.fallbackRegistries.map(registry =>
+                    this.activeConnection.call({
+                        method: "GET",
+                        path: client.VERSION + "libpod/images/search",
+                        body: "",
+                        params: {
+                            term: registry + "/" + value
+                        }
+                    })));
+            }
         }
 
         Promise.allSettled(searches)
                 .then(reply => {
                     if (reply && this._isMounted) {
                         let imageResults = [];
-                        let dialogError = "";
-                        let dialogErrorDetail = "";
+                        const manifestResult = handleManifestsQuery(reply[0]);
 
-                        for (const result of reply) {
+                        for (let i = 1; i < reply.length; i++) {
+                            const result = reply[i];
                             if (result.status === "fulfilled") {
                                 imageResults = imageResults.concat(JSON.parse(result.value));
-                            } else {
-                                dialogError = _("Failed to search for new images");
-                                // TODO: add registry context, podman does not include it in the reply.
-                                dialogErrorDetail = result.reason ? cockpit.format(_("Failed to search for images: $0"), result.reason.message) : _("Failed to search for images.");
+                            } else if (!manifestResult && !imageExistsLocally(value, this.props.localImages)) {
+                                changeDialogError(result.reason);
```

Comment:
So this overwrites the error set by `handleManifestQuery()`, which seems fine -- but it also overwrites earlier error from reply[i-1]. It probably doesn't matter that much if we show the error from the first or last registry, it just feels a bit random.

Reply:
yes, it's going to show the last error encountered. It's the same behavior as before